scopesim/optics/optical_train.py

```diff
@@ -370,6 +370,15 @@ def readout(self, filename=None, **kwargs):
         - Apply detector plane (0D, 2D) effects - z_order = 500..599
 
         """
+
+        # Hack to make sure AutoExposure and Quantization work properly.
+        # Should probably be removed once #428 is fixed properly.
+        if kwargs.get('exptime', None) is not None:
+            if "!OBS.dit" in self.cmds:
+                self.cmds.pop("!OBS.dit")
+            if "!OBS.ndit" in self.cmds:
+                self.cmds.pop("!OBS.ndit")
```

Comment:
```suggestion
            self.cmds.pop("!OBS.dit", None)
            self.cmds.pop("!OBS.ndit", None)
```

Providing a default saves the `if ... in ...` check, without any negative consequences that a know of.

Reply:
For real? Pop accepts a default? When did that happen?? [Apparently this has always been the case](https://docs.python.org/3/whatsnew/2.3.html) since the introduction of `pop()` in Python 2.3. Thanks.